node/pkg/db/governor.go

```diff
@@ -308,36 +310,28 @@ func (d *Database) GetChainGovernorDataForTime(logger *zap.Logger, now time.Time
 			}
 
 			if IsPendingMsg(key) {
-				p, err := UnmarshalPendingTransfer(val)
+				p, err := UnmarshalPendingTransfer(val, false)
 				if err != nil {
 					return err
 				}
 
-				if time.Until(p.ReleaseTime) > maxEnqueuedTime {
-					p.ReleaseTime = now.Add(maxEnqueuedTime)
-					err := d.StorePendingMsg(p)
-					if err != nil {
-						return fmt.Errorf("failed to write new pending msg for key [%v]: %w", p.Msg.MessageIDString(), err)
-					}
+				pending = append(pending, p)
+			} else if isOldPendingMsg(key) {
+				p, err := UnmarshalPendingTransfer(val, true)
+				if err != nil {
+					return err
 				}
 
 				pending = append(pending, p)
+				oldPendingToUpdate = append(oldPendingToUpdate, p)
```

Comment:
do we need to do anything in the handling of the old msgs in this slice? like set `IsReobservation` field or at least make the raw msg bytes longer?

Reply:
The marshaled object has the new field added, defaulted to false. We have to assume false on anything reloaded.

Adding it to this slice causes us to store the message in the new format and delete the one in the old format, which is all we need to do.